mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp

```diff
@@ -52,6 +52,39 @@ struct GPUToSPIRVPass final : impl::ConvertGPUToSPIRVBase<GPUToSPIRVPass> {
   bool mapMemorySpace;
 };
 
+spirv::TargetEnvAttr
+spirvTargetWithSpatialExtents(spirv::TargetEnvAttr in,
+                              gpu::SpatialExtentsAttr spatialExtents) {
+  auto oldResources = in.getResourceLimits();
+
+  unsigned subgroupSize = oldResources.getSubgroupSize();
+  std::optional<unsigned> minSubgroupSize = oldResources.getMinSubgroupSize();
+  std::optional<unsigned> maxSubgroupSize = oldResources.getMaxSubgroupSize();
+  DenseI32ArrayAttr maxComputeWorkgroupSize =
+      cast<DenseI32ArrayAttr>(oldResources.getMaxComputeWorkgroupSize());
```

Comment:
```suggestion
  ArrayAttr maxComputeWorkgroupSize = oldResources.getMaxComputeWorkgroupSize();
```

No need to cast below